lib/transformer.js

```diff
@@ -0,0 +1,159 @@
+'use strict'
+
+const {promisify} = require('util')
+const zlib = require('zlib')
+const aws = require('aws-sdk')
+const {
+  checkFileFormat
+, formatObjectKey
+, hasProperty
+} = require('./utils.js')
+
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+module.exports = exports = {
+  extractData
+, getLogs
+, prepareLogs
+, getObject: promisify(s3.getObject)
+}
+
+function extractData(data) {
+  if (!data || !Object.keys(data).length) return []
+
+  const {
+    Records
+  , logFiles
+  , ...meta
+  } = data
+
+  const base = {
+    timestamp: Date.now()
+  }
+
+  let records = data
+  if (Records) {
+    records = Records
+    base.meta = meta
+  } else if (logFiles) {
+    records = logFiles
+    base.meta = meta
+  }
+
+  if (meta && meta.eventTime) {
+    base.timestamp = (new Date(meta.eventTime)).getTime()
+  }
+
+  if (!Array.isArray(records)) {
+    const lineJSON = {
+      line: records
+    , ...base
+    }
+
+    if (typeof records !== 'string') {
+      lineJSON.line = JSON.stringify(records)
+    }
+
+    return [lineJSON]
+  }
+
+  return records.map((record) => {
+    const lineJSON = {
+      line: record
+    , ...base
+    }
+
+    if (typeof record !== 'string') {
+      lineJSON.line = JSON.stringify(record)
+    }
+
+    if (record.eventTime) {
+      lineJSON.timestamp = (new Date(record.eventTime)).getTime()
+    }
+
+    return lineJSON
+  })
+}
+
+async function getLogs(params) {
+  let err
+  if (!hasProperty(params, 'Key') || !hasProperty(params, 'Bucket')) {
+    err = new Error('Both Bucket and Key params must be provided')
+    err.meta = {params}
+    throw err
+  }
+
+  let data
+  try {
+    data = await exports.getObject(params)
+  } catch (error) {
+    err = new Error(`Error in Getting the Object: ${error}`)
+    err.meta = {error, params}
+    throw err
```

Comment:
When making your own error, consider adding more context.  Seeing just this error in the logs wouldn't tell us where the problem is.  By using `meta` we can give more context.  Also, it's not necessary to include the `error` in the `message` string when it's going into `meta`.  If you look at the current test for this, the error messages are a bit redundant, and that's why.  How about something like this:

```suggestion
    err = new Error('Error during getLogs')
    err.meta = {
      message: 's3.getObjects threw an error'
    , error
    , params
    }
    throw err
```

Reply:
Done!